### PR TITLE
user resource: add plan modifier

### PR DESCRIPTION
**Pull request by stuart-mclaren-hpe**
This plan modifier ensures that if `password_wo` is missing during
plan it is detected and an error is raised.

`password_wo` is a corner case where it is required for create, but
not for import. We do not raise an error in the import case
(tested by existing test).

### CODE REVIEW

File: internal/subproviders/morpheus/modifiers/modifiers.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// RequireOnCreateModifier can be used for corner cases where
+// an attribute is optional for import, but required for create
+type RequireOnCreateModifier struct{}
+
+func (m RequireOnCreateModifier) Description(_ context.Context) string {
+	return "Requires the attribute to be set during resource creation."
+}
+
+func (m RequireOnCreateModifier) MarkdownDescription(ctx context.Context) string {
+	return m.Description(ctx)
```
[the remark this answers is not in the document]
Comment (stuart-mclaren-hpe) [reply internal/subproviders/morpheus/modifiers/modifiers.go:20]: We're using the same text for both, so line 20 is just calling line 15

**Review by jasonquigleyhpecom — APPROVED** [review]

**Review by jasonquigleyhpecom — APPROVED** [review]
